feat(text-to-image):using-gemini

shivdvaio opened the pull request.

HirogaKatageri commented on sparrow-clients/lib/src/prompts/text_to_image.dart:
```diff
@@ -0,0 +1,70 @@
+
+Future<void> main() async {
+
+  await dotenv.load();
```
There's no need to use dotenv, since we can access an environment variable using Platform.env[]

If you're attempting to test locally, my recommendation is to simply create a file e.g.

```
export GOOGLE_API_KEY=<YOUR_KEY_HERE>
```

In your terminal session proceed with source ./<the_path_to_the_file>

shivdvaio replied:
got it

HirogaKatageri commented on sparrow-clients/lib/src/prompts/text_to_image.dart:
```diff
@@ -0,0 +1,70 @@
+import 'package:path/path.dart' as path;
+import 'package:sparrow_clients/sparrow-clients.dart';
+
+Future<void> main() async {
```
main() means this is an application or an executable. Which should not be the case in prompts. If this is an example, kindly move this to examples.

shivdvaio commented:
This PR adds support for Gemini's text-to-image generation capabilities to the sparrow-clients library.

## Changes
- Added `textToImage` method to `GeminiClient` class
- Extended `GeminiPrompt` to handle image generation requests
- Added example demonstrating text-to-image generation (`example/text_to_image.dart`)
- Added support for base64-encoded image output
- Implemented proper error handling

## Implementation Details
- Uses Gemini's `gemini-2.0-flash-preview-image-generation` model
- Returns base64-encoded image data that can be saved as PNG
- Maintains consistent API style with existing `textPrompt` method
- Handles response parsing and error cases

## Example Usage
```dart
final client = GeminiClient(apiKey: Platform.environment['GEMINI_API_KEY']!);
final base64Image = await client.textToImage(prompt: "A serene landscape");
await File('generated_image.png').writeAsBytes(base64Decode(base64Image));
```

## Testing
- Added example in `sparrow-clients/example/text_to_image.dart`
- Tested with various prompts and verified image generation
- Maintains error handling consistent with existing functionality

## Notes
- Requires valid Gemini API key with image generation access
- Generated images are saved in PNG format
- Response includes base64-encoded image data
